hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenBurnHandler.java

```diff
@@ -137,8 +136,7 @@ public void handle(@NonNull final HandleContext context) throws HandleException
                     tokenRelStore);
 
             // Update treasury's NFT count
-            final var treasuryAcct =
-                    accountStore.get(asAccount(token.treasuryAccountId().accountNum()));
+            final var treasuryAcct = accountStore.get(token.treasuryAccountId());
```

Comment:
@Neeharika-Sompalli I don't see a check in burn's validate semantics for a usable treasury account. Should we be calling `getIfUsable` here?

Reply:
@mhess-swl  yes, doesn't hurt to use `getIfUsable`